framework/data/ActiveDataProvider.php

```diff
@@ -103,7 +103,9 @@ protected function prepareModels()
         }
         $query = clone $this->query;
         if (($pagination = $this->getPagination()) !== false) {
-            $pagination->totalCount = $this->getTotalCount();
+            if (($pagination->totalCount = $this->getTotalCount()) === 0) {
```

Comment:
why put all the logic in one line? Good practice is that every action goes to a separate line. Here you are doing assignment and then comparison on one ...

I can't see any performance optimization neither on this syntax.